### PR TITLE
[wallet-ext] refactor the importing of Ledger accounts to use useQuery

### PR DESCRIPTION
**Pull request by williamrobertson13**
## Description 
This PR refactors the importing of Ledger accounts to use `useQuery` which is a little more ergonomic than using `useEffect`. In the rush of meeting last week's deadline, I overlooked that we can use `useQuery` for any type of async data retrieval operation (the "server" in this case is the Ledger device). This reduces some of the boilerplate code for loading/error states and when caching is possible at some point, we'll get it out of the box for free.

## Test Plan 
- Manual testing (initial connection, cached re-connection, disconnection during the middle of the flow, etc.)

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [X] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes
N/A

### CODE REVIEW

File: apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx [hunk 1]
```diff
@@ -29,16 +31,34 @@ export function ImportLedgerAccounts() {
...
+            );
+            return ledgerAccounts.filter(({ address }) => {
+                return !existingAccountAddresses.includes(address);
+            });
+        },
+        onError: (error) => {
+            if (error instanceof LockedDeviceError) {
+                toast.error('Your device is locked. Un-lock it and try again.');
+            } else {
+                toast.error('Make sure you have the Sui application open.');
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx:57]: `Make sure you have the Sui application on your Ledger device` to be a bit more verbose?

File: apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx [hunk 2]
```diff
@@ -29,16 +31,34 @@ export function ImportLedgerAccounts() {
...
+        data: ledgerAccounts,
+        isLoading: areLedgerAccountsLoading,
+        isError: encounteredDerviceAccountsError,
+    } = useDeriveLedgerAccounts({
+        numAccountsToDerive: numLedgerAccountsToDeriveByDefault,
+        select: (ledgerAccounts) => {
+            const existingAccountAddresses = existingAccounts.map(
+                (account) => account.address
+            );
+            return ledgerAccounts.filter(({ address }) => {
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx:49]: could skip the additional `map`
```
ledgerAccounts.filter(({address}) => !existingAccounts.some(account => account.address === address))
```

**Review by mamos-mysten — APPROVED** [review]

File: apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx [hunk 3]
```diff
@@ -29,16 +31,34 @@ export function ImportLedgerAccounts() {
...
+                return !existingAccountAddresses.includes(address);
+            });
+        },
+        onError: (error) => {
+            if (error instanceof LockedDeviceError) {
+                toast.error('Your device is locked. Un-lock it and try again.');
```
[the remark this answers is not in the document]
Comment (williamrobertson13) [reply apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx:55]: thx (grammar is hard)

File: apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx (hunk 2 above)
Comment (williamrobertson13) [reply apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx:49]: thx

**Pull request merged**